### PR TITLE
Feature/role listener

### PR DESCRIPTION
**Pull request by earayu**

### CODE REVIEW

File: go/vt/vttablet/tabletmanager/tm_init.go [hunk 1]
```diff
@@ -881,3 +889,19 @@ func (tm *TabletManager) initializeReplication(ctx context.Context, tabletType t
...
+func (tm *TabletManager) syncVTTabletType(ctx context.Context, lastUpdate time.Time, targetTabletType topodatapb.TabletType) (bool, error) {
+	within30Seconds := func() bool {
+		return time.Since(lastUpdate) <= 30*time.Second
+	}
+
+	if tm.Tablet().Type == targetTabletType && within30Seconds() {
+		return false, nil
+	}
+	log.Infof("start to change vttablet role to %s", targetTabletType.String())
+	err := tm.ChangeType(ctx, targetTabletType, false)
```
Comment (newborn22) [inline go/vt/vttablet/tabletmanager/tm_init.go:902]: When `tm.Tablet().Type == targetTabletType `, why should we still run tm.ChangeType? Curious about the logic here.

File: go/vt/vtconsensus/vtconsensus.go [hunk 2]
```diff
@@ -162,6 +164,9 @@ func (vtconsensus *VTConsensus) RefreshCluster() {
 			shard.KeyspaceShard.Keyspace, shard.KeyspaceShard.Shard, refreshInterval.Seconds())
 		ticker := time.Tick(refreshInterval)
 		for range ticker {
+			if !enableVtconsensus {
```
[the remark this answers is not in the document]
Comment (earayu) [reply go/vt/vtconsensus/vtconsensus.go:167]: No, we don't run the goroutine, we will return here

File: go/vt/vttablet/tabletmanager/tm_init.go (hunk 1 above)
Comment (earayu) [reply go/vt/vttablet/tabletmanager/tm_init.go:902]: We will force apply `ChangeType` every 30 seconds, to prevent the ABA problem

**Review by newborn22 — APPROVED** [review]
LGTM

**Pull request merged**